packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2213,17 +2241,25 @@ export class SnapController extends BaseController<
       await this.#assertIsInstallAllowed(snapId, {
         version: newVersion,
         checksum: manifest.source.shasum,
-        permissions: manifest.initialPermissions,
+        initialPermissions: manifest.initialPermissions,
+        dynamicPermissions: manifest.dynamicPermissions,
       });
 
-      const processedPermissions = processSnapPermissions(
+      // Process Initial and Dynamic permissions
+      const processedInitialPermissions = processSnapPermissions(
         manifest.initialPermissions,
       );
-
-      this.#validateSnapPermissions(processedPermissions);
+      this.#validateSnapPermissions(processedInitialPermissions);
+      const processedDynamicPermissions = processSnapPermissions(
+        manifest.dynamicPermissions ?? {},
+      );
```

Comment:
We should validate dynamic permissions too

Reply:
Answered in: https://github.com/MetaMask/snaps/pull/2088#discussion_r1514600397